config/config.go

```diff
@@ -216,6 +220,9 @@ func (c *Config) Validate() error {
 			return fmt.Errorf("invalid subnetID in configuration. error: %v", err)
 		}
 		sourceBlockchainIDs = append(sourceBlockchainIDs, blockchainID)
+
+		// Write back to the config
+		c.SourceSubnets[i] = s
```

Comment:
This was an existing bug. `c.SourceSubnets` is a slice of values, not references, so updating `s` (as occurs in `s.Validate`) did not update `c`. We only updated `s` to populate the allowed destinations list, which was not used anywhere in our tests.